features/step_wrapper_with_options.feature

```diff
@@ -0,0 +1,42 @@
+Feature: Step Wrapper with Options
+  In order to be able to write more complex step definition wrappers
+  As a developer
+  I want Cucumber to provide the "options" object to the wrapping function
+
+  Background:
+    Given a file named "features/a.feature" with:
+      """
+      Feature: Step with an option
+        Scenario: Steps
+          When I run a step with options
+      """
+    And a file named "features/step_definitions/cucumber_steps.js" with:
+      """
+      import {defineSupportCode} from 'cucumber'
+
+      defineSupportCode(({Then, When}) => {
+        When(/^I run a step with options$/, {retry: 2}, function () {
```

Comment:
I'm hesitant about this as it means options can now be filled with non-standard options. Thoughts on putting the the wrapper options in a specific child `wrapperOptions`. Thus this would become
```js
When(/^I run a step with options$/, {wrapperOptions: {retry: 2}}, function () {
```
And in `setDefinitionFunctionWrapper` it is passed the `fn` and `options.wrapperOptions`